File: leetcode/0162-find-peak-element/solution.py
class Solution:
    def findPeakElement(self, nums: List[int]) -> int:
        return self.findPeakElementUtil(nums, 0, len(nums) - 1)
    
    def findPeakElementUtil(self, nums: List[int], lo: int, hi: int) -> int:
        if lo == hi:
            return lo;
        
        mid = lo + (hi - lo) // 2
        if nums[mid] < nums[mid + 1]:
            return self.findPeakElementUtil(nums, mid + 1, hi);
        return self.findPeakElementUtil(nums, lo, mid);

File: leetcode/0162-find-peak-element/solution.cpp
class Solution {
public:
    int findPeakElement(vector<int>& nums) {
        return findPeakElement(nums, 0, nums.size() - 1);
    }
    
    int findPeakElement(vector<int>& nums, int lo, int hi) {
        if (lo == hi)
            return lo;
        
        int mid = lo + (hi - lo) / 2;
        if (nums[mid] < nums[mid + 1])
            return findPeakElement(nums, mid + 1, hi);
        return findPeakElement(nums, lo, mid);
    }
};